react-js/react-part-2.md

```diff
@@ -0,0 +1,219 @@
+
+   ```
+    <input />
+    <button type="button" >Add</button>
+   ```
+
+3. Now we will create state object of the component and assign properties to it.
+
+#### Note :- All the input box defined in React are controlled by state. In other words the value of input box should be assigned to a property of state object.
+
+4. So, there will be a array of items and value as property in state which is assigned as input box value.
+
+   ```
+    state = {
+      items: [],
+      value:""
+    }
+   ```
+
+   ```
+   <input value = {this.state.value} />
+   ```
+
+5. A function must be declared so that when the input box value changes the state should be updated. Let us define a function `onChangeHandler` which will update the state assigned to onChange event handler of input box.
+
+   ```
+   onChangeHandler = (event) => {
+   this.setState({
+     value: [event.target.value]
+   })
+   }
+   ```
+
+   ```
+   <input value = {this.state.value} onChange={this.onChangeHandler} />
+   ```
+
+6. Another function to add the items in the array when `add` button is clicked. This function will be assigned to onClick event handler of add button.
+
+   ```
+   addItem = () => {
+     this.setState({
+       value: "",
+       items: [...this.state.items,this.state.value]
+       })
+    }
+   ```
+
+   ```
+   <button type="button" onClick={this.addItem} >Add</button>
+   ```
+
+7. Now we will render the array items using `map` function.
+
+   ```
+   <ol>
+     {this.state.items.map((item,index) => (
+       <li key={index}>
+        {item}
+       </li>
+      ))}
+    </ol>
+   ```
+
+8. Finally your `App.js` will look like -
+
+    ```
+    import React from 'react';
+    import './App.css';
+
+    class App extends React.Component {
+     state = {
+       items: [],
+       value:""
+     }
+
+    // Function to handle input value of the text field
+     onChangeHandler = (event) => {
+       this.setState({
+         value: [event.target.value]
+       })
+     }
+
+    // Function to add new values to items array
+     addItem = () => {
+       this.setState({
+         value: "",
+         items: [...this.state.items,this.state.value]
+       })
+     }
+
+     render() {
+       return (
+         <div className="App">
+           <input value={this.state.value} onChange={this.onChangeHandler} />
+           <button type="button" onClick={this.addItem} >Add</button>
+           <ol>
+             {this.state.items.map((item,index) => (
+               <li key={index}>
+                 {item}
+               </li>
+              ))}
+            </ol>
+          </div>
+        );
+      }
+    }
+
+    export default App;
+    ```
+
+### Independent Practice
+
+- Create an increment/decrement counter which includes click functionality. On click of increment button adds 1 to the value and decrement button will subtract the number by 1. Initially the number will be 0.
```

Comment:
Since this is now the guided practice, please come up with another one for independent practice. A list of items would be good, as shown above.